### ISSUE TITLE
Support for embededding/transcluding whole or parts of other notes

### ISSUE THREAD

**Issue by fabiospampinato** [notable/notable#1029]

**Comment by ThanosPal** [comment]
This would be extremely useful and powerful. Maybe using the `#` notation to transclude a specific header and its contents? 

example: 
`![](@note/Foo.md#some-header-inside-note)`

**Comment by devopsfreude** [comment]
Recently began trying out TiddlyWiki for this very feature. Double curly braces `{{}}` are are used for transclusion, but only includes an entire tiddler (note). RoamResearch uses double parenthesis `(())` to include specific lines from another note. I like Notable because it is not subscription cloud based and with transclusion ... nirvana!

**Comment by AmberHaelo** [comment]
There is in fact a popular Markdown flavour, MultiMarkdown, that uses `{{/path/to/file}}` notation for transclusion, which would lend further weight to supporting that use. With the addition of the @note marker to substitute for the archive path, dual system compatibility could be achieved on the MultiMarkdown side with a pre-processor that replaces `@note` for `/full/path/to/place/notes/`.

I'm not as sure about taking existing Markdown syntax for images and using that to mean and do something else entirely; what happens if you put something into the place where the caption/alt text goes--what does that mean? How is transclusion semantically similar to figure placement outside of the sheer mechanical technology involved with interleaving bytes into output? Down that path, I feel strangeness lies.

**Comment by fabiospampinato** [comment]
> I'm not as sure about taking existing Markdown syntax for images and using that to mean and do something else entirely;

I see that as generalizing the syntax, usually `!` before a link means "that's an image", a generalization of that would be "rendering this thing", which makes sense IMHO.

**Comment by AmberHaelo** [comment]
> I see that as generalizing the syntax, usually `!` before a link means 
> "that's an image", a generalization of that would be "rendering this 
> thing", which makes sense IMHO.

Yeah, I get that, and it's what I meant by there being a mechanical 
similarity: of inserting this thing, or rendering this thing. I'm 
thinking about it more from the semantic or humanist standpoint though, 
how is text assembly similar to a figure or even link environment, which 
has need for such conventions as caption/hyperlink text?

In terms of presentation, the result of text assembly is usually meant 
to be invisible to the reader, it isn't in that sense a semantic 
declaration like a heading is, or a figure, which in and of itself 
communicates to the reader with its forms, the intention of the content. 
It's a lower level technical *tool* that is only visible to the authors, 
like comments are.

Well, it's maybe Markdown-manifesto stuff to be thinking that way. Not 
everything in the Markdown ethos is applicable to the utility of its use 
inside of software!

**Comment by fabiospampinato** [comment]
@AmberHaelo You lost me there at some point. I think the similarity lies in the "rendering something" output. Alternatively something like this: `![[other note]]` will work too if you like that better.

**Comment by a-ruskomaa** [comment]
This would be a great addition! Some documentation tools use @@include token to achieve this. This would be somewhat in line with the existing syntax (@note, @tag, etc).

Possible usage: @@include[Optional label](@note/filename.md)

Whatever the chosen syntax is, I think the user should be able to specify if they want the embedded note/snippet to be visually separated (using gray background?) from the rest of the note or not. Maybe the content should be embedded "as is" if the label is not specified and visually separated (with a labeled link to the original note?) if it is.

**Comment by a-ruskomaa** [comment]
This would be a great addition! Some documentation tools use ```@@include``` token to achieve this. This would be somewhat in line with the existing syntax (```@note```, ```@tag```, etc).

Possible usage: ```@@include[Optional label](@note/filename.md)```

Whatever the chosen syntax is, I think the user should be able to specify if they want the embedded note/snippet to be visually separated (using gray background?) from the rest of the note or not. Maybe the content should be embedded "as is" if the label is not specified and visually separated (with a labeled link to the original note?) if it is.
